### DISCUSSION TITLE
/post/none_to_0_dot_1

### DISCUSSION THREAD

**Discussion by swilcox** [swilcox/swilcox.github.io#4] [General]
https://swilcox.github.io/post/none_to_0_dot_1/ 

 defining your version number in one place.

**Comment by bcorfman** [comment]
I'm somewhat embarrassed that I've never delved into this issue. In some of my projects, version numbers proliferate as you describe, and I've sort of suffered with it. I'm going to bookmark this and come back to your solution with some of my longer-lived projects.

**Comment by bcorfman** [comment]
I ran across a doc page on this topic on the [Python Packaging User Guide](https://packaging.python.org/en/latest/guides/single-sourcing-package-version/#single-sourcing-the-version). 
I like dynamic = ["version"]. 
Do you have any added opinions?

**Reply by bcorfman** [reply]
Related: https://stackoverflow.com/questions/72270892/git-versioning-with-setuptools-in-pyproject-toml

**Reply by swilcox** [reply]
Thanks for the links. I have updated the post to include a whole section on this method and I've also linked to a PR on a project where I implemented this.
